ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -585,6 +606,15 @@ private FunctionDescriptionList describeNonAggregateFunction(
     );
   }
 
+  private CommandStatusEntity distributeCreateAsSelect(
```

Comment:
The `addInferredSchema` call in the normal `distributeStatement` call only actually does anything for `CT` and `CS` statements.  So you _could_ follow the same pattern and just add `withInferredSinkTopic` into that.

Of course, via the handler map we _know_ that we've got a CSAS or CTAS as so can be more selective. Which is the other way of doing things.

TBH, not sure which I prefer.  However, having two patterns isn't ideal:

1. existing: infer method takes untyped `PreparedStatement` and just does a no-op for some types. KsqlResource calls the infer method for all distributed statements.
1. adding in this pr: infer method takes specific type, so caller must determine when to call it. Hence KsqlResource needs new handler map entry, which calls infer and then distributes.

I think I probably prefer the existing pattern, as it makes the calling code simpler.  `StatementUtil` could just have a single `addInferredInfo` to it, which _does the right thing_ regardless of the type of statement passed to it.  This way, the functionality is encapsulated in the  util class, rather than bleeding through to the calling class. 

I'm not sure we gain anything from the type safety of the new pattern you've added in this PR.  It just means more code in KsqlResource.

What do you think?

Reply:
I'm ambivalent, though I agree we should only have one pattern. If you slightly prefer the existing I'm happy to refactor to do that :-)